python/golem_task_api/testutils/tasklifecycleutil.py

```diff
@@ -0,0 +1,172 @@
+import shutil
+from pathlib import Path
+from typing import Callable, List, Tuple
+
+from async_generator import asynccontextmanager
+
+from golem_task_api import (
+    TaskApiService,
+    constants,
+    ProviderAppClient,
+    RequestorAppClient,
+)
+
+
+class TaskLifecycleUtil:
+    def __init__(self, work_dir: Path) -> None:
+        print('workdir:', work_dir)
+        self.work_dir = work_dir
+        self.req_work_dir = work_dir / 'requestor'
+        self.req_work_dir.mkdir()
+        self.prov_work_dir = work_dir / 'provider'
+        self.prov_work_dir.mkdir()
+
+    def init_provider(
+            self,
+            get_task_api_service: Callable[[Path], TaskApiService],
```

Comment:
I think we can make this interface a little bit more convenient. I believe `TaskApiService` is something that app developers shouldn't even know about. This method should accept a `ProviderHandler` instead. To make it possible `InlineTaskApiService` should be moved here as well and made a little bit more generic so that it could be used for arbitrary provider classes.

Reply:
Fair enough, I already have a draft with moving InlineService into this package as well.

Reply:
But I mean, we can provide the InlineService implementation, maybe even docker one too, but these methods still need to take the service not just the handler. Otherwise, how would you decide which service to use? Blenderapp uses both inline and docker for tests.

Reply:
I'd say only inline should be used here. Docker (and other 'real' environments) should be part of the integration test utils suite.

Reply:
So you're saying that blenderapp shouldn't test docker integration? I dunno, I find it valuable.

Reply:
I find it valuable as well and I'm not saying it shouldn't be tested at all.
But if we put Docker-based TaskApiService here, then what about other environments? Are we going to add a new test class here for every environment we introduce into Golem? I think task API should stay environment-agnostic. And so should these test utils. Docker integration should be tested with the "integration test utils".

Reply:
But making this method take just the handler won't be compatible with blenderapp using docker for tests - which was my initial point.

Reply:
OK, so maybe for now we can have two methods? One accepting a `ProviderHandler` (and using `InlineTaskApiService`) and the other accepting a `TaskApiSerivce` factory function.

Reply:
Done